src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -462,8 +462,28 @@ private void CloseInnerConnection()
 
         override public void Close()
         {
+            ConnectionState previousState = State;
+            Guid operationId;
+            Guid clientConnectionId;
+
+            // during the call to Dispose() there is a redundant call to 
```

Comment:
If the close call is actually redundant, should the beginning of the method just check the state and exit immediately if it's not open?